### ISSUE TITLE
Enhancement: `allowDirectConstAssertionInArrowFunctions` should not error when using `satisfies`

### ISSUE THREAD

**Issue by jleider** [typescript-eslint/typescript-eslint#10230]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

eslint-plugin

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Description

I have a situation where I want to use both `as const` followed by `satisfies`. With the rule currently as written as soon as you add a `satisfies` after an `as const` the rule will error. Leaving off the `satisfies` works as expected. 

For example given `allowDirectConstAssertionInArrowFunctions: true`

```typescript

type Foo = {
  foo: string,
  bar: number,
};

const a = () => ({
  foo: "FOOO",
  bar: 123,
} as const); // <-- This works as expected.

const a = () => ({
  foo: "FOOO",
  bar: 123,
} as const satisfies Foo); // <-- This throws an error.

```

In this scenario, the `satisfies Foo` provides extra type safety ensuring that `a satisfies Foo` but still keeping the ability to have explicit type literals for `foo: "FOOO"` and `bar: 123`.

**Comment by kirkwaiblinger** [comment]
Hi @jleider! Could you please fill out the issue template? It request helpful information like 
- what rule this is about
- a reproduction on [our playground](https://typescript-eslint.io/play)
that save us time and help make the issue actionable. Hint - if you use the playground, there is a "report as issue" button that will even fill out most of the template for you. Thanks!

**Comment by jleider** [comment]
@kirkwaiblinger Thank you, I have created a new bug https://github.com/typescript-eslint/typescript-eslint/issues/10231 using the template from the playground.

**Comment by kirkwaiblinger** [comment]
Thanks! it is hugely appreciated! 🙂
